src/main/java/io/nats/client/impl/NatsMessage.java

```diff
@@ -26,103 +25,78 @@
     private String sid;
     private String subject;
     private String replyTo;
-    private byte[] data;
-    private byte[] protocolBytes;
+    private ByteBuffer data;
+    private ByteBuffer protocolBytes;
     private NatsSubscription subscription;
-    private long sizeInBytes;
-    
-    NatsMessage next; // for linked list
-
-    static final byte[] digits = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9'};

```

Comment:
This isn't needed, should be faster to just add `0x30` to convert an integer to ascii.

Reply:
add to the PR, should be ok - although i might - since i am paranoid - convert 0 to a byte to get the value :-)

Reply:
ASCII "0" to byte gives you `0x30`, I'm doing this in integer mode then casting to `byte` when writing to the buffer, at least I think that's probably the most efficient way to do it.